tests/test_scan.py

```diff
@@ -365,3 +368,109 @@ def test_scan_execute_with_soda_server_client_scan_result_does_not_contain_any_e
     )
 
     assert not scan_result.has_errors()
+
+
+def test_test_results_to_data_frame(spark_session: SparkSession) -> None:
+    """Test conversions of test_result to dataframe."""
+    expected = spark_session.createDataFrame(
+        [
+            {
+                "test": Test(
+                    id="id",
+                    title="title",
+                    expression="expression",
+                    metrics=["metrics"],
+                    column="column",
+                ),
+                "passed": True,
+                "skipped": False,
+                "values": {"value": "10"},
+                "error": "exception",
+                "group_values": {"group": "by"},
+            }
+        ]
+    ).select("error", "group_values", "passed", "skipped", "test", "values")
```

Comment:
I would prefer not to hard code the columns. Is the order of the columns the problem? Or is the intent to select a subset of the columns?

If it is the first, then I would not do anything here. And change the `.select` below to `.select(*expected.columns)`

Reply:
The order of the columns is the problem.

Reply:
This line has resolved the issue with order of columns.
`assert expected.select(sorted(expected.columns)).collect() == out.select(sorted(out.columns)).collect()`